index.html

```diff
@@ -198,12 +198,16 @@ <h2>Step 5</h2>
   var i = locationReplacements.length; while (i--) {
       location = location.replace(locationReplacements[i][0], locationReplacements[i][1]);
   }
-  var splitLocation = location.split("/");
-  if (document.getElementById("clayton-addresses").checked && splitLocation[0].startsWith("CL_") && claytonStreetNames.hasOwnProperty(splitLocation[0].substr(-3, 3))) {
-    location = "Room " + splitLocation[1] + ", " + splitLocation[0].slice(3, -3) + " " + claytonStreetNames[splitLocation[0].substr(-3, 3)];
-  }
-  if (obj.location === "-") {
-    location = "";
+
+  if (location.startsWith("CL_")) {
+    const [_, streetCode, streetNumber, buildingName, roomCode] = location.match(/CL_(\w{3})-(\d{1,2})\.(.+)_(.+)/);
+    if (document.getElementById("clayton-addresses").checked && location.startsWith("CL_") && claytonStreetNames.hasOwnProperty(streetCode)) {
+      // Escape commas to ensure compatibility with macOS Calendar
+      location = `Room ${roomCode}\\, ${buildingName}\\, ${streetNumber} ${claytonStreetNames[streetCode]}`;
```

Comment:
Does this result in backslashes in other clients? If so, we should move this as an option.

Reply:
I actually took a look at this and [RFC5545 section 3.3.11](https://tools.ietf.org/html/rfc5545#section-3.3.11) says that commas should always be escaped in properties of type TEXT inside an ICS file:

> A COMMA character in a "TEXT" property value MUST be escaped with a BACKSLASH character.

The `Location` property is a TEXT value, so if a client isn't escaping it, it's not following spec. FWIW, I have tested this with Google Calendar and it is correctly handling the escaped `Location` property, if that's of any reassurance.

Reply:
SGTM